libs/live-nft-react/src/NftMetadataProvider.tsx

```diff
@@ -117,6 +117,46 @@ export function useNftCollectionMetadata(
   }
 }
 
+export function useNftCollectionMetadataBatch(
+  items: Array<{
+    contract: string | undefined;
+    currencyId: string | undefined;
+  }>,
+): Array<NFTResource<NonNullable<NFTCollectionMetadataResponse["result"]>>> {
+  const { cache, loadCollectionMetadata } = useContext(NftMetadataContext);
+
+  const hasAllProperties = (item: { contract?: string; currencyId?: string }) =>
+    item.contract && item.currencyId;
+
+  const data = items.map(item => {
+    const key = hasAllProperties(item) ? getNftCollectionKey(item.contract!, item.currencyId!) : "";
+    const cachedData = cache[key] as NFTResource<
+      NonNullable<NFTCollectionMetadataResponse["result"]>
+    >;
+    return { key, cachedData };
+  });
+
+  useEffect(() => {
+    data.forEach(({ key, cachedData }) => {
+      const item = items.find(
+        i => hasAllProperties(i) && getNftCollectionKey(i.contract!, i.currencyId!) === key,
+      );
+      if (!item) return;
+      if (!cachedData || isOutdated(cachedData)) {
+        loadCollectionMetadata(item.contract!, item.currencyId!);
+      }
+    });
+  }, [items, loadCollectionMetadata, cache, data]);
+
+  return items.map(item => {
```

Comment:
IMO hooks are expected to return memoized values, which can be safely used as dependencies without creating unnecessary re-renders. So I think the returned value should also be wrapped in a `useMemo`